src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -927,7 +800,7 @@ def post_synth_sdc(self) -> Optional[str]:
         pass
 
 
-class CadenceTool(HasSDCSupport, HammerTool):
+class CadenceTool(HasSDCSupport, HasCPFSupport, HasUPFSupport, HammerTool):
```

Comment:
Why `HasUPFSupport`?

Reply:
The cadence tools do support UPF, so this is future proofing for when someone implements the generation for it.